frontend/src/composables/useShootSpec.js

```diff
@@ -119,11 +121,20 @@ export function useShootSpec (shootItem, options = {}) {
   })
 
   const isCustomShootDomain = computed(() => {
-    return some(shootDnsProviders.value, 'primary')
+    return !!shootDnsPrimaryProvider.value
   })
 
-  const shootDnsProviders = computed(() => {
-    return get(shootSpec.value, 'dns.providers')
+  const shootDnsPrimaryProvider = computed(() => {
+    return find(shootSpec.value.dns?.providers, { primary: true })
+  })
+  const shootExtensionDnsProviders = computed(() => {
```

Comment:
should this computed property not just return the providers instead of also mapping the `secretRefName` into it? It's only used in one place, maybe it reduces complexity if we look up the `secretRefName` at the place where we need it?

Reply:
Yes, tis is kind of a leftover. I learned about this resources secretRef thingy only after fist implementation. Moved it to GShootInfrastructureCard